File: java-back-server/src/main/java/space/borisgk/taxi/api/model/entity/AuthService.java
package space.borisgk.taxi.api.model.entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Table(name = "auth_service")
@Getter
@AllArgsConstructor
public class AuthService {
    // TODO инициальзация при старте приложения через рефлексию
    public final static AuthService VK = new AuthService(1, "VK"), FB = new AuthService(2, "FB");

    private AuthService(){}

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "name")
    private String name;

    public static AuthService fromName(String name) {
        switch (name) {
            case "VK":
                return VK;
            case "FB":
                return FB;
        }
        return null;
    }
}

File: kotlin-back-server/src/main/kotlin/space/borisgk/taxi/api/model/entity/AuthService.kt
package space.borisgk.taxi.api.model.entity

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class AuthService {

    constructor(id: Int?) {
        this.id = id
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
        set(id) {
            field = this.id
        }
    var name: String? = null
        set (name) {
            field = this.name
        }
}